File: ruby/ranks.rb
# https://www.codewars.com/kata/545f05676b42a0a195000d95/train/ruby

def ranks(scores)
  ranks = scores.map.with_index.sort_by { |score, index| -score }
  p ranks
  rank_map = {}
  
  ranks.each_with_index do |(score, index), rank|
    rank_map[score] ||= rank + 1
  end

  scores.map { |score| rank_map[score] }
end

File: javascript/ranks.js
function ranks(scores) {
    const ranks = scores
        .map((score, index) => [score, index])
        .sort((a, b) => b[0] - a[0]);
    
    const rankMap = {};
    
    ranks.forEach(([score], rank) => {
        if (!(score in rankMap)) {
            rankMap[score] = rank + 1;
        }
    });

    return scores.map(score => rankMap[score]);
}
